ktor-client/ktor-client-core/common/src/io/ktor/client/features/HttpRedirect.kt

```diff
@@ -39,9 +70,18 @@ class HttpRedirect {
                     url.parameters.clear()
 
                     location?.let { url.takeFrom(it) }
+
+                    if (call.response.status.isRewritable()) {
+                        if (feature.rewritePostAsGet && call.request.method == HttpMethod.Post)
+                            method = HttpMethod.Get
+                    }
                 })
+                redirects++
 
                 if (!call.response.status.isRedirect()) return call
+
+                if (redirects >= feature.maximumRedirects)
+                    throw RedirectCountExceedException("Max redirect count ${feature.maximumRedirects} exceeded")
```

Comment:
It's quite ambiguous with a limit in the `HttpSend` feature - we throw 2 kinds of exceptions for similar reasons.
Could you tell me if you have an idea of how we can simplify the handling of such cases?